### ISSUE TITLE
Rust plugin in Rider 2023.2 doesn't work properly anymore.

### ISSUE THREAD

**Issue by En3Tho** [intellij-rust/intellij-rust#10826]
## Working Environment

* IntelliJ Rust plugin version: 0.4.200.5421-231
* IntelliJ Toml plugin version: 231.8109.91
* Rust toolchain version: 1.72.0
* IDE name and version: Rider 2023.1.4
* Operating system: Windows 11

![image](https://github.com/intellij-rust/intellij-rust/assets/37334640/bae376cf-4d23-4871-8840-0fca8374327c)

## Failing environment:

* IntelliJ Rust plugin version: 0.4.200.5421-232
* IntelliJ Toml plugin version: 232.8660.88
* Rust toolchain version: 1.72.0
* IDE name and version: Rider 2023.2.1
* Operating system: Windows 11

![image](https://github.com/intellij-rust/intellij-rust/assets/37334640/55fc3536-e092-47bf-9670-83c0d6a10c96)

## Problem description

After updating to 2023.2 I'm not able to get any Rust intellisense or whatever anymore. Attaching cargo toml doesn't work.
I tried to:
Clean everything up and reattach from the clean slate.
Uninstall-install Rust plugin.
I tried to attach "target" directory (see message below)

Nothing did help.

I'm able to find this SEVERE message in the idea log which is surprising
```
2023-08-25 11:33:31,137 [  28957] SEVERE - #org.rust.stdext.AsyncValue - java.lang.RuntimeException: java.lang.IllegalStateException: Exclude folder file://G:/source/repos/En3Tho/RustPlayground/RustWindowsService/target must be under content entry file://G:/source/repos/En3Tho/RustPlayground/.run
java.util.concurrent.CompletionException: java.lang.RuntimeException: java.lang.IllegalStateException: Exclude folder file://G:/source/repos/En3Tho/RustPlayground/RustWindowsService/target must be under content entry file://G:/source/repos/En3Tho/RustPlayground/.run
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:98)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.codeWithMe.ClientId$Companion.decorateFunction$lambda$8(ClientId.kt:321)
	at com.intellij.codeWithMe.ClientId$Companion.decorateFunction$lambda$8(ClientId.kt:321)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

## Steps to reproduce

I have a very small playground project: https://github.com/En3Tho/RustPlayground

Basically run it with 2023.1 Rider then with 2023.2. I don't know if it is Rider specific, maybe it's just an overall Idea problem.

**Comment by Maksim20023** [comment]
Hi! Thank you for the report.
The issue may appear in the project with multiple packages that aren't combined into a [workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html). Using a workspace is optional but desirable to avoid problems. So, please try to place your project into the [Cargo workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) and let me know if it helped.

**Comment by En3Tho** [comment]
Thank you. I will try.

Technically those projects are not the part of the same workspace though. They don't share dependencies or output directory, only the same root directory.
